src/ripple/shamap/README.md

```diff
@@ -26,164 +56,292 @@ There are two different ways of building and using a SHAMap:
  1. A mutable SHAMap and
  2. An immutable SHAMap
 
-The distinction here is not of the classic C++ immutable-means-unchanging
-sense.  An immutable SHAMap contains *nodes* that are immutable.  Also,
-once a node has been located in an immutable SHAMap, that node is
-guaranteed to persist in that SHAMap for the lifetime of the SHAMap.
+The distinction here is not of the classic C++ immutable-means-unchanging sense.
+ An immutable SHAMap contains *nodes* that are immutable.  Also, once a node has
+been located in an immutable SHAMap, that node is guaranteed to persist in that
+SHAMap for the lifetime of the SHAMap.
 
-So, somewhat counter-intuitively, an immutable SHAMap may grow as new nodes
-are introduced.  But an immutable SHAMap will never get smaller (until it
-entirely evaporates when it is destroyed).  Nodes, once introduced to the
-immutable SHAMap, also never change their location in memory.  So nodes in
-an immutable SHAMap can be handled using raw pointers (if you're careful).
+So, somewhat counter-intuitively, an immutable SHAMap may grow as new nodes are
```

Comment:
If a new node is introduced, then its parent's digest must change, and so on up all the way to the root. Does that mean that when a node is introduced, the `SHAMap` points to a new root?

Reply:
For immutable `SHAMap`s, the hashes never change.  When a new node is added, it is because the node exists, it is just missing from the trie.  The node is not a new account state or transaction.  It will be found in a cache, local database or on the p2p network.

Reply:
Ah! This might be one of the big a-ha moments I needed for the rest of this to click. A mutable `SHAMap` can add new nodes, while an immutable `SHAMap` can only fill in blanks. It wasn't clear to me before that an immutable `SHAMap` could have blanks (i.e. missing nodes). It might be explained further down, but I stopped reading once I was too confused. Maybe we can introduce this idea in the section where we explain mutability?

Reply:
See the section titled **Mutability** just below the node diagram (all committed just an hour ago).  If that doesn't do it for you, I'm open to suggestions.  And thanks for the other wording suggestions you made.  They should all be included in this update.